src/crimson/os/seastore/transaction_manager.h

```diff
@@ -305,6 +305,64 @@ class TransactionManager : public ExtentCallbackInterface {
     });
   }
 
+  /**
+   * remap_pin
+   *
+   * Remap original extent to new extent.
+   * Return the pin of new extent.
+   */
+  using remap_pin_iertr = base_iertr;
+  using remap_pin_ret = remap_pin_iertr::future<LBAMappingRef>;
+  template <typename T>
+  remap_pin_ret remap_pin(
+    Transaction &t,
+    LBAMappingRef &&pin,
+    extent_len_t remap_offset,
+    extent_len_t remap_len) {
+    LOG_PREFIX(TransactionManager::remap_pin);
+    auto remap_laddr = pin->get_key() + remap_offset;
+    auto remap_paddr = pin->get_val().add_offset(remap_offset);
+    // FIXME: paddr can be absolute and pending
+    ceph_assert(pin->get_val().is_absolute());
+    ceph_assert(remap_offset < pin->get_length());
+    ceph_assert(remap_offset + remap_len <= pin->get_length());
```

Comment:
ceph_assert that remap_len cannot be 0.

Also ceph_assert that remap_offset/len must be epm->get_block_size() aligned.